internal/api/graphql/access/auth.go

```diff
@@ -17,23 +19,52 @@ type Logger interface {
 	Warn(...interface{})
 }
 
-type Auth interface {
-	GetMiddleware() gin.HandlerFunc
+func NewAuth(cfg *util.Config) *Auth {
+	l := newLogger()
+	auth := Auth{logger: l}
+	auth.AppendInstance(NewTokenAuthMethod(l, cfg))
+	//TODO: auth.AppendInstance(NewOidcAuthMethod(l, cfg))
+	return &auth
 }
 
-func NewAuth(cfg *util.Config) Auth {
-	l := newLogger()
+type Auth struct {
+	chain  []AuthMethod
```

Comment:
Maybe I'm missing the full picture, but why exactly do we need a chain of auth methods?

Reply:
Hmmm.. Even if OIDC uses JWTs as part of its specification, we do need to treat both auth mechanisms differently. I mean, you also have different fields (inside the JWT token payload, see this [example](https://connect2id.com/learn/openid-connect#:~:text=The%20ID%20token%20statements%2C%20or%20claims%2C%20are%20packaged%20in%20a%20simple%20JSON%20object%3A)). Or how is this supposed to work? Maybe we need some call to clarify this.